cmd/metrics-collector/main.go

```diff
@@ -55,6 +55,7 @@ func main() {
 
 	// TODO: more complex input definition, such as a JSON struct
 	cmd.Flags().StringArrayVar(&opt.Rules, "match", opt.Rules, "Match rules to federate.")
+	cmd.Flags().StringArrayVar(&opt.RecordingRules, "recordingrule", opt.RecordingRules, "Define recording rules to reduce metrics before sending them, one rule per line.")
```

Comment:
recording rule is to generate new metrics based on specified query expression. 

" reduce metrics before sending them" is the one purpose we can use recording rule to achieve